coalib/core/Bear.py

```diff
@@ -179,24 +198,19 @@ def maintainers_emails(cls):
                 if cls.MAINTAINERS_EMAILS == set() else
                 cls.MAINTAINERS_EMAILS)
 
-    @enforce_signature
-    def __init__(self, section: Section, file_dict: dict):
+    @staticmethod
+    def setup_dependencies():
         """
-        Constructs a new bear.
-
-        :param section:
-            The section object where bear settings are contained.
-        :param file_dict:
-            The file-dictionary containing a mapping of filenames to the
-            according file contents.
-        :raises RuntimeError:
-            Raised when bear requirements are not fulfilled.
+        This is a user defined function that can download and set up
+        dependencies (via download_cached_file or arbitrary other means) in an
+        OS independent way.
         """
-        self.section = section
-        self.file_dict = file_dict
-
-        self._dependency_results = defaultdict(list)
 
+    def __attrs_post_init__(self):
+        """
+        This function is called after all attributes are initialized.
+        It checks for pre-requisites.
+        """
```

Comment:
As `__attrs_post_init__` is known how to work for `attrs`, you could omit the docstring. But up to you :)